Editorial: fix usage of processResponseConsumeBody

nicolo-ribaudo opened the pull request:
Ref https://github.com/whatwg/html/issues/7996. ~I didn't update `synchronous perform the fetch` steps, because I couldn't find what's the signature of _Fetch_ when using it synchronously.~

TODO:
- [x] Update step 3 of https://w3c.github.io/manifest/#dfn-processing-a-manifest to use `bodyBytes` instead of trying to read `response`'s body. https://github.com/w3c/manifest/pull/1058
- [x] Update the SW spec to use the single "perform the fetch" hook type


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://whatpr.org/html/8412/infrastructure.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">/infrastructure.html</a>  ( <a href="https://whatpr.org/html/8412/b9a961d...094a101/infrastructure.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">diff</a> )
<a href="https://whatpr.org/html/8412/links.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">/links.html</a>  ( <a href="https://whatpr.org/html/8412/b9a961d...094a101/links.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">diff</a> )
<a href="https://whatpr.org/html/8412/semantics.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">/semantics.html</a>  ( <a href="https://whatpr.org/html/8412/b9a961d...094a101/semantics.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">diff</a> )
<a href="https://whatpr.org/html/8412/webappapis.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">/webappapis.html</a>  ( <a href="https://whatpr.org/html/8412/b9a961d...094a101/webappapis.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">diff</a> )
<a href="https://whatpr.org/html/8412/workers.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">/workers.html</a>  ( <a href="https://whatpr.org/html/8412/b9a961d...094a101/workers.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">diff</a> )
<a href="https://whatpr.org/html/8412/worklets.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">/worklets.html</a>  ( <a href="https://whatpr.org/html/8412/b9a961d...094a101/worklets.html" title="Last updated on Oct 31, 2022, 1:50 AM UTC (094a101)">diff</a> )

nicolo-ribaudo commented on source:
```diff
@@ -14777,8 +14787,9 @@ interface <dfn interface>HTMLLinkElement</dfn> : <span>HTMLElement</span> {
-     <var>success</var>, and <var>response</var>.</li>
+     <!-- TODO: We should probably pass bytesOrNull -->
+     <li><p><span>Process the linked resource</span> given <var>el</var>, <var>success</var>, and
+     <var>response</var>.</li>
```
Should I update step 3 of https://w3c.github.io/manifest/#dfn-processing-a-manifest to use `bytesOrNull` passed from here?

domenic replied:
That would be lovely!

The other usage site is https://html.spec.whatwg.org/#link-type-stylesheet:process-the-linked-resource , which has an XXX box about how it's failing to use the bytes... I guess update the XXX box to reference the passed-in _bytesOrNull_?

domenic commented on source:
```diff
@@ -14750,24 +14750,34 @@ interface <dfn interface>HTMLLinkElement</dfn> : <span>HTMLElement</span> {
     data-x="processResponseConsumeBody">processResponseConsumeBody</i> set to the following steps
-    given <span data-x="concept-response">response</span> <var>response</var>:</p>
+    given <span data-x="concept-response">response</span> <var>response</var> and null, failure, or
+    a <span>byte sequence</span> <var>bytesOrNull</var>:</p>
```
I tend to prefer the name "bodyBytes" for this parameter? WDYT?

nicolo-ribaudo replied:
I copied `bytesOrNull` from somewhere else, but it's not 100% correct (because it can also be _failure_): `bodyBytes` sounds ok, with the implicit knowledge (which always explicitly checked anyway) that it could be null/failure.

domenic commented on source:
```diff
@@ -93176,14 +93194,14 @@ document.querySelector("button").addEventListener("click", bound);
     <p>Otherwise, <span data-x="concept-fetch">fetch</span> <var>request</var> with <i
-    data-x="processResponseConsumeBody">processResponseConsumeBody</i> set to
-    <var>processResponse</var> as defined below.</p>
+    data-x="processResponseConsumeBody">processResponseConsumeBody</i> as defined below.</p>
```
```suggestion
    data-x="processResponseConsumeBody">processResponseConsumeBody</i> set to <var>processResponseConsumeBody</var> as defined below.</p>
```

(We want the phrasing to work even if we hadn't named the variable the same as the parameter.)

domenic commented on source:
```diff
@@ -93445,11 +93461,12 @@ document.querySelector("button").addEventListener("click", bound);
-  settings object</var>, <var>destination</var>, <var>credentials mode</var>, <var>module map
-  settings object</var>, <var>onComplete</var>, and the following <span>asynchronous perform
-  the fetch hook</span> given <var>request</var> and <i
+  object</var>, a <span data-x="concept-Worklet-module-responses-map">module responses map</span>
```
We shouldn't link to a field definition as if it were a type definition. So I think just omit the type here.

domenic commented on source:
```diff
@@ -93896,14 +93925,14 @@ document.querySelector("button").addEventListener("click", bound);
-    <p>Otherwise, <span data-x="concept-fetch">fetch</span> <var>request</var> with
-    <var>processResponse</var> as <i
-    data-x="processResponseConsumeBody">processResponseConsumeBody</i>.</p>
+    <p>Otherwise, <span data-x="concept-fetch">fetch</span> <var>request</var> with <i
+    data-x="processResponseConsumeBody">processResponseConsumeBody</i> as defined below.</p>
```
```suggestion
    <p>Otherwise, <span data-x="concept-fetch">fetch</span> <var>request</var> with <i
    data-x="processResponseConsumeBody">processResponseConsumeBody</i> set to <var>processResponseConsumeBody</var> as defined below.</p>
```

domenic commented on source:
```diff
@@ -104678,82 +104705,88 @@ interface <dfn interface>SharedWorkerGlobalScope</dfn> : <span>WorkerGlobalScope
      <li><p>If <var>isTopLevel</var> is false, <span data-x="concept-fetch">fetch</span>
-     <var>request</var> with <var>processCustomFetchResponse</var> as <i
-     data-x="processResponseConsumeBody">processResponseConsumeBody</i> and return.</p></li>
+     <var>request</var> with <i data-x="processResponseConsumeBody">processResponseConsumeBody</i>
```
What is processResponseConsumeBody set to? Probably processCustomFetchResponse?

domenic commented on source:
```diff
@@ -104678,82 +104705,88 @@ interface <dfn interface>SharedWorkerGlobalScope</dfn> : <span>WorkerGlobalScope
-     algorithm, and run the remaining steps as part of the fetch's <span>process response</span> for
-     the <span data-x="concept-response">response</span> <var>response</var>.</p></li>
+     <li>
+      <p><span data-x="concept-fetch">Fetch</span> <var>request</var> with with <i
```
with with -> with

domenic reviewed (commented):
Thank you as always!

nicolo-ribaudo commented:
I added a commit to also fix the signature for synchronous fetching, by removing synchronous fetching (since it's not supported anymore) and pausing while waiting for the result of a normal fetch.

Since there is now a single type of fetch, I reverted part of https://github.com/whatwg/html/pull/8328: there is now again a single "perform the fetch" hook type, which accepts a callback that takes response and bytes. This is exactly how it works for the default fetch.

domenic reviewed (commented):
Only a few minor nits. Thanks so much, again!! Ping the thread when service workers and manifest are ready to go. (Or if you decide you'd rather let the maintainers of those specs update themselves, which is totally valid; you've done so much already.)

nicolo-ribaudo commented:
Done! I also updated https://github.com/w3c/ServiceWorker/pull/1658, since I already had that PR open, and I would appreciate also a review on that side even if you are not the maintainer of that spec (but the changes are very similar to the changes in this PR).
